sys/include/net/nanocoap.h

```diff
@@ -0,0 +1,600 @@
+ * @returns         <0 on error
+ */
+ssize_t coap_reply_simple(coap_pkt_t *pkt,
+        unsigned code,
+        uint8_t *buf, size_t len,
+        unsigned ct,
+        const uint8_t *payload, uint8_t payload_len);
+
+/**
+ * @brief   Handle incoming CoAP request
+ *
+ * This function will find the correct handler, call it and write the reply
+ * into @p resp_buf.
+ *
+ * @param[in]   pkt             pointer to (parsed) CoAP packet
+ * @param[out]  resp_buf        buffer for response
+ * @param[in]   resp_buf_len    size of response buffer
+ *
+ * @returns         size of reply packet on success
+ * @returns         <0 on error
+ */
+ssize_t coap_handle_req(coap_pkt_t *pkt, uint8_t *resp_buf, unsigned resp_buf_len);
+
+/**
+ * @brief   Builds a CoAP header
+ *
+ * Caller *must* ensure @p hdr can hold the header and the full token!
+ *
+ * @param[out]   hdr        hdr to fill
+ * @param[in]    type       CoAP packet type (e.g., COAP_TYPE_CON, ...)
+ * @param[in]    token      token
+ * @param[in]    token_len  length of @p token
+ * @param[in]    code       CoAP code (e.g., COAP_CODE_204, ...)
+ * @param[in]    id         CoAP request id
+ *
+ * @returns      length of resulting header
+ */
+ssize_t coap_build_hdr(coap_hdr_t *hdr, unsigned type, uint8_t *token, size_t token_len, unsigned code, uint16_t id);
+
+/**
+ * @brief   Insert a CoAP option into buffer
+ *
+ * This function writes a CoAP option with nr. @p onum to @p buf.
+ * It handles calculating the option delta (from @p lastonum), encoding the
+ * length from @p olen and copying the option data from @p odata.
+ *
+ * @param[out]  buf         buffer to write to
+ * @param[in]   lastonum    number of previous option (for delta calculation),
+ *                          or 0 for first option
+ * @param[in]   onum        number of option
+ * @param[in]   odata       ptr to raw option data (or NULL)
+ * @param[in]   olen        length of @p odata (if any)
+ *
+ * @returns     amount of bytes written to @p buf
+ */
+size_t coap_put_option(uint8_t *buf, uint16_t lastonum, uint16_t onum, uint8_t *odata, size_t olen);
+
+/**
+ * @brief   Insert content type option into buffer
+ *
+ * @param[out]  buf             buffer to write to
+ * @param[in]   lastonum        number of previous option (for delta
+ *                              calculation), or 0 if first option
+ * @param[in]   content_type    content type to set
+ *
+ * @returns     amount of bytes written to @p buf
+ */
+size_t coap_put_option_ct(uint8_t *buf, uint16_t lastonum, uint16_t content_type);
+
+/**
+ * @brief   Insert URI encoded option into buffer
+ *
+ * @param[out]  buf         buffer to write to
+ * @param[in]   lastonum    number of previous option (for delta calculation),
+ *                          or 0 if first option
+ * @param[in]   uri         ptr to source URI
+ * @param[in]   optnum      option number to use (e.g., COAP_OPT_URI_PATH)
+ *
+ * @returns     amount of bytes written to @p buf
```

Comment:
Although `@returns` and `@return` do the same thing, at least inside a single file we should be consistent in the usage... Further check the indention after the `@return(s)` statements, we have also two styles in this file...